source/common/network/io_socket_handle_impl.h

```diff
@@ -12,13 +12,29 @@
 namespace Envoy {
 namespace Network {
 
+/**
+ * On different platforms the sockaddr struct for unix domain
+ * sockets is different. We use this function to get the
+ * length of the platform specific struct.
+ */
+static constexpr socklen_t udsAddressLength() {
+#if defined(__APPLE__)
```

Comment:
@wrowe @sunjayBhatia @mattklein123 this is the place that I intended to use the `platform` enum.

This could have been written with a `constexpr if (current_platform == platformTypes::Windows` instead of `#ifdefs`.

Reply:
Seeing this example, and other situations it could be applied, I could see this pattern working, or just pulling this into a define in platform.h

Since this function is only used in one place, might be worth moving into the file it is used in an anonymous namespace like helper functions are in other places

Reply:
I moved the helper function into `io_socket_handle_impl.cc` in an anonymous namespace in 783c656ed4c27cb06f855164af1765b648fe706c.

The idea that I had in mind was to have the platform.h for types and the rest to be moved near the place that they are used.